test/AllPairs/test_ap.py

```diff
@@ -41,12 +44,13 @@ def test_allpairs_singularity(self):
         out_path = Path(OUT_DIR+'sample-out.txt')
         out_path.unlink(missing_ok=True)
         # Only include required arguments and run with Singularity
+        config.config.framework = "singularity"
```

Comment:
This does look awkward. An alternative would be having a different config file for testing purposes, but then we would have to maintain it as new algorithms are added.

Reply:
I agree that having a separate testing config that needs to be updated/tracked sounds like the wrong approach, especially when we're trying to increase how many PRMs we ingest.

Another option would be to keep the container framework as an input to each PRM's run method. Then in the tests we can just do:
```
AllPairs.run(
            container_framework="singularity"
            nodetypes=TEST_DIR+'input/sample-in-nodetypes.txt',
            network=TEST_DIR+'input/sample-in-net.txt',
            output_file=str(out_path)
        )
```